src/containers/ExperimentContainer.tsx

```diff
@@ -68,31 +74,66 @@ export const ExperimentContainer = () => {
   // Get the experiment object and the index of the current module.
   const dispatch = useDispatch()
   const experiment = useSelector(experimentSelector)
+  const experimentModules = useSelector(allModulesSelector)
   const currentModule = useSelector(currentModuleSelector)
 
   // If the user has been 'screened out' then show respective screen
   if (experiment.participantRejected) {
     return <RejectionScreen onExit={() => terminateExperiment(false)} />
   }
 
+  // If all modules have been completed...
+  if (experiment.currentModuleIndex === experiment.definition?.modules.length) {
+    // Mark experiment complete
+    if (!experiment.isComplete)
+      dispatch(
+        updateExperiment({
+          ...experiment,
+          isComplete: true,
+        }),
+      )
+
+    // Are all the modules synced?
+    if (!experiment.offlineOnly) {
+      return (
+        <SummaryScreen
+          modules={experimentModules}
+          syncExperiment={() => dispatch(syncExperiment)}
+          onExit={() => terminateExperiment(false)}
+        />
+      )
+    } else {
+      terminateExperiment(false)
+    }
+
+    // Render nothing...
+    return null
+  }
+
   // If no experiment saved then return the login
   if (experiment.definition == undefined || currentModule == undefined) {
     return <LoginScreen />
   }
 
-  // If all modules rendered then go back to login
-  if (experiment.currentModuleIndex === experiment.definition.modules.length) {
-    terminateExperiment(false)
-    return null
-  }
-
   // Get the current module data and it's respective component.
   const ModuleComponent: ExperimentModule =
     ExperimentModuleTypes[currentModule?.moduleType]
 
   // This function is called when the experiment should transition to next module.
   function onModuleComplete() {
+    // Mark current module as complete
+    dispatch(
+      updateModule({
+        ...currentModule,
+        moduleCompleted: true,
+      }),
+    )
+
+    // Proceed to next module
     dispatch(nextModule())
+
+    // Sync past screen data
+    dispatch(syncExperiment)
```

Comment:
Does this need to be called? `dispatch(syncExperiment())`?

Reply:
Nope, You pass the function and redux thunk calls it with paramaters `(dispatch, getState)`.